cpp/src/gandiva/simple_arena.h

```diff
@@ -0,0 +1,127 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef GANDIVA_SIMPLE_ARENA_H
+#define GANDIVA_SIMPLE_ARENA_H
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+namespace gandiva {
+
+#ifdef GDV_HELPERS
```

Comment:
Is this still necessary? This is a new API, backwards compatibility isn't required.